File: src/routes/datasets.js
import { Router } from "express";
import { join } from "path";
import { getProjectPath } from "../utils/paths.js";

const router = Router();

// Environment variables for paths (all relative to project root)
const DATA_DIR = process.env.DATA_DIR
    ? getProjectPath(process.env.DATA_DIR)
    : getProjectPath("data");

const DATASETS = ["comment_generation", "code_refinement"];

router.get("/download/:dataset", async (req, res) => {
    const { dataset } = req.params;
    const withContext = req.query.withContext
        ? JSON.parse(req.query.withContext)
        : false;

    if (!DATASETS.includes(dataset)) {
        return res.status(400).json({ error: "Invalid dataset name" });
    }

    const fileName = `${dataset}_${withContext ? "with_context" : "no_context"}.zip`;
    const filePath = join(DATA_DIR, fileName);

    try {
        res.download(filePath);
    } catch (error) {
        console.error("Error serving file:", error);
        res.status(500).json({ error: "Error serving file" });
    }
});

export default router;

File: src/routes/datasets.py
# routes/datasets.py
from flask import Blueprint, send_from_directory, request, jsonify
import os

router = Blueprint('datasets', __name__, url_prefix='/datasets')

DATASETS = {'comment_generation', 'code_refinement'}

# below, the '../' + is need because the send_from send_from_directory is local
# the file, but the DATA_DIR is local to the root of the repo
DATA_DIR = os.path.join('..', os.getenv("DATA_PATH", "data"))


@router.route('/download/<dataset>')
def download(dataset):
    if dataset not in DATASETS:
        return jsonify({'error': 'Invalid dataset name'}), 400
    with_ctx = request.args.get('withContext', 'false').lower() == 'true'
    fname = f"{dataset}_{'with_context' if with_ctx else 'no_context'}.zip"
    return send_from_directory(DATA_DIR, fname, as_attachment=True)
